File: bun-cli/src/handlers/list.ts
/**
 * List notes handler using Effect.
 */

import { Effect } from "effect";
import pc from "picocolors";
import { NoteServiceTag } from "../service/note-service";
import { printNotesTable } from "../ui/table";

/**
 * Execute list command.
 */
export const execute = Effect.gen(function* () {
  console.log(pc.cyan("\n📋 Fetching notes..."));

  const service = yield* NoteServiceTag;
  const notes = yield* service.list();

  if (notes.length === 0) {
    console.log(pc.yellow("📭 No notes found."));
    return;
  }

  console.log(pc.green(`\nFound ${notes.length} note(s)\n`));
  printNotesTable(notes);
});

File: backend/crates/cli/src/handlers/list.rs
//! List notes handler.

use anyhow::Result;
use colored::Colorize;

use db::{NoteRepository, NoteService};

use crate::ui::table;

/// Execute the list notes command.
pub async fn execute<R: NoteRepository>(service: &NoteService<R>) -> Result<()> {
    println!("\n{}", "📋 Fetching notes...".bright_blue());

    let notes = service.list().await?;

    if notes.is_empty() {
        println!(
            "\n{}",
            "📭 No notes found. Create your first one!".bright_yellow()
        );
    } else {
        println!(
            "\n{} {} note(s)\n",
            "Found".bright_green(),
            notes.len().to_string().bright_white().bold()
        );
        table::display_notes(&notes);
    }

    Ok(())
}
